modules/utils/src/eth.ts

```diff
@@ -20,10 +26,21 @@ export const getGasPrice = async (provider: Provider, providedChainId?: number):
   return chainId === 100 && price.lt(minGasPrice) ? minGasPrice : price;
 };
 
-export const hydrateProviders = (chainProviders: ChainProviders): HydratedProviders => {
-  const hydratedProviders: { [url: string]: JsonRpcProvider } = {};
+/// Parse CSV formatted provider dict into ChainRpcProviders, which uses a list of Urls per chainId.
+export const parseProviders = (prevChainRpcProviders: { [chainId: string]: string }): ChainRpcProviders => {
+  var chainProviders: ChainRpcProviders = {}
+  Object.entries(prevChainRpcProviders).forEach(
+    ([chainId, urlString]) => {
+      chainProviders[chainId] = urlString.split(",");
+    }
+  );
+  return chainProviders
+}
+
+export const hydrateProviders = (chainProviders: ChainRpcProviders): HydratedProviders => {
+  const hydratedProviders: { [url: string]: ChainRpcProvider } = {};
   Object.entries(chainProviders).map(([chainId, url]) => {
-    hydratedProviders[chainId] = new StaticJsonRpcProvider(url as string, parseInt(chainId));
+    hydratedProviders[chainId] = new ChainRpcProvider(parseInt(chainId), url);
```

Comment:
Noting that we replaced StaticJsonRpcProvider here - is there something special / different about StaticJsonRpcProvider vs JsonRpcProvider? Do we need to implement additional functionality or a 'Static' version of ChainRpcProvider?

Reply:
Hm, we switched to `StaticJsonRpcProvider` because it cuts down on the number of RPC calls. We should just instantiate as `StaticJsonRpcProvider`s under the hood. Is that possible?

Reply:
Yes but lower priority atm

Reply:
Do you mean replace JsonRpcProvider with StaticJsonRpcProvider entirely? Or just in this one use case?

Reply:
Yes replace entirely. That's what we already did AFAIK.